lib/runtime/version.go

```diff
@@ -97,6 +99,48 @@ func (v *LegacyVersionData) TransactionVersion() uint32 {
 	return 0
 }
 
+// Encode returns the SCALE encoding of the Version
+func (v *LegacyVersionData) Encode() ([]byte, error) {
+	type Info struct {
```

Comment:
This can be replaced with an anonymous struct.
```go
	info := &struct {
		SpecName         []byte
		ImplName         []byte
		AuthoringVersion uint32
		SpecVersion      uint32
		ImplVersion      uint32
	}{
		SpecName:         v.specName,
		ImplName:         v.implName,
		AuthoringVersion: v.authoringVersion,
		SpecVersion:      v.specVersion,
		ImplVersion:      v.implVersion,
	}
```